File: app/src/main/java/com/example/dkalev/remember/model/DecksDatabase.java
package com.example.dkalev.remember.model;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

@Database(entities = {Deck.class, Card.class}, version = 1)
public abstract class DecksDatabase extends RoomDatabase{

    private static volatile DecksDatabase INSTANCE;

    public abstract DeckDao deckDao();

    public static DecksDatabase getInstance(Context context) {
        if (INSTANCE == null) {
            synchronized (DecksDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            DecksDatabase.class, "Decks.db")
                            .build();
                }
            }
        }
        return INSTANCE;
    }
}

File: app/src/main/java/com/example/dkalev/remember/model/DecksDatabase.kt
package com.example.dkalev.remember.model

import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase
import android.arch.persistence.room.TypeConverters
import com.example.dkalev.remember.model.card.Card
import com.example.dkalev.remember.model.card.CardDao
import com.example.dkalev.remember.model.deck.Deck
import com.example.dkalev.remember.model.deck.DeckDao

@Database(entities = [(Deck::class), (Card::class)], version = 1)
@TypeConverters(DateTypeConverter::class)
abstract class DecksDatabase : RoomDatabase() {

    abstract fun deckDao(): DeckDao
    abstract fun cardDao(): CardDao

}